server/controllers/admin/placementRequests/bookingsController.test.ts

```diff
@@ -52,22 +62,32 @@ describe('PlacementRequestsController', () => {
 
       await requestHandler(request, response, next)
 
-      expect(response.render).toHaveBeenCalledWith('admin/placementRequests/bookings/new', {
-        pageHeading: 'Record an Approved Premises (AP) placement',
-        premises,
-        placementRequest,
-        errors: {},
-        errorSummary: [],
-        errorTitle: undefined,
-        isWomensApplication: false,
-        ...DateFormats.isoDateToDateInputs(placementRequest.expectedArrival, 'arrivalDate'),
-        ...DateFormats.isoDateToDateInputs('2022-01-15', 'departureDate'),
-      })
+      expect(response.render).toHaveBeenCalledWith('admin/placementRequests/bookings/new', expectedRenderParameters)
 
       expect(premisesService.getCas1All).toHaveBeenCalledWith(token, { gender: 'man' })
       expect(placementRequestService.getPlacementRequest).toHaveBeenCalledWith(token, placementRequest.id)
     })
 
+    it('should filter out premises that operate space-bookings rather than legacy bookings', async () => {
+      ;(fetchErrorsAndUserInput as jest.Mock).mockReturnValue({ errors: {}, errorSummary: [], userInput: {} })
+      ;(placementRequest.application as ApprovedPremisesApplication).isWomensApplication = false
+
+      const mixedPremises: Array<Cas1PremisesBasicSummary> = [
+        ...cas1PremisesBasicSummaryFactory.buildList(2, { supportsSpaceBookings: true }),
+        ...cas1PremisesBasicSummaryFactory.buildList(2, { supportsSpaceBookings: false }),
+      ]
+      premisesService.getCas1All.mockResolvedValue(mixedPremises)
+
+      const requestHandler = bookingsController.new()
+
+      await requestHandler(request, response, next)
+
+      expect(response.render).toHaveBeenCalledWith('admin/placementRequests/bookings/new', {
+        ...expectedRenderParameters,
+        premises: mixedPremises.filter(({ supportsSpaceBookings }) => !supportsSpaceBookings),
+      })
+    })
```

Comment:
Given this is a very focused test, the assert would maybe benefit from using the `expect.objectContaining()` approach:

```suggestion
      expect(response.render).toHaveBeenCalledWith('admin/placementRequests/bookings/new', expect.objectContaining({
        premises: mixedPremises.filter(({ supportsSpaceBookings }) => !supportsSpaceBookings),
      })
    })
```

Reply:
Yeah, makes sense.